File: php-solutions/1-twosum.php
<?php

class Solution{
    function twosum($nums, $target){
        $map =[];
        for ($i=0; $i<count($nums); $i++){
            $residue = $target - $nums[$i];
            if (array_key_exists($residue, $map)){
                return [$i, $map[$residue]];
            }else{
                $map[$nums[$i]] = $i;
            } 
        }
        throw new InvalidArgumentException('No two sum solution');
    }
}

$sol = new Solution();

$nums = [2, 7, 11, 15];

$target = 9;

print_r($sol->twosum($nums, $target));

File: go-solutions/1-twosum.go
package main

import "fmt"

func twosum(nums []int, target int) []int {
	m := make(map[int]int)
	for i := 0; i < len(nums); i++ {
		residue := target - nums[i]
		val, ok := m[residue]
		if ok {
			return []int{val, i}
		} else {
			m[nums[i]] = i
		}
	}
	return []int{}
}

func main() {
	fmt.Print(twosum([]int{2, 3, 11, 7}, 9))
}
